graph/edge/impl/TypeEdgeImpl.java

```diff
@@ -407,8 +408,8 @@ public static class Persisted extends TypeEdgeImpl implements TypeEdge {
         private final AtomicBoolean deleted;
         private TypeVertex from;
         private TypeVertex to;
-        private VertexIID.Type overriddenIID;
-        private TypeVertex overridden;
+        private final AtomicBoolean committed;
+        private Either<VertexIID.Type, TypeVertex> overridden;
```

Comment:
I'm struggling to come up with a more readable solution.
Here, `Either.first` only ever holds the existing persisted IID (or null). When it's requested or modified, `Either.second` is set to the corresponding `TypeVertex` (or null).

This does mean commit will write if the overridden was requested (but not modified) - which is presumably rare since we were lazy about loading `TypeVertex overridden` from `overriddenIID`.